docs: promote use of bill of materials in quickstart documentation (#1620)
googleapis/google-auth-library-java: README.md

## Quickstart

> ### Using Maven

Reviewer: Thanks!

Thoughts on also promoting libraries-bom here (perhaps a small section)? We can mention that if you use libraries-bom, then auth-bom is imported as well. Otherwise, you should manually import the bom (like below)

Author: Sounds good. I'll add a section.

Author: Ah, I think it's not as trivial to ensure the snippet with libraries-bom has an up-to-date `<version/>` since it's not managed in versions.txt.

Other repos such as google-cloud-java use the [config yaml](https://github.com/googleapis/google-cloud-java/blob/7d99222b94664eb74edb5c592f85ce05bf9bfdff/generation_config.yaml#L3).
We also have the case of spring-cloud-gcp which seems to [rely on renovate-bot](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/renovate.json). I think it's better to wait for such PR and otherwise raise an issue to keep it up to date.

Reviewer: Ah yeah makes sense. Maybe we don't need to have a code reference here. We could just reference this page: https://cloud.google.com/java/docs/bom

#### Google Auth Library Bill of Materials
In order to ensure transitive dependencies and the modules themselves are aligned with each other,
we rely on the Google Auth Library Bill of Materials. Please add this to your dependency management
section as follows:

[//]: # ({x-version-update-start:google-auth-library-bom:released})
```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>com.google.auth</groupId>
      <artifactId>google-auth-library-bom</artifactId>

>       <version>1.30.1</version>

Reviewer: Can this version be updated by release-please?

Reviewer: Oh, I see the tag above it (`# ({x-version-update-start:google-auth-library-bom:released})`). If this is already auto updated, feel free to resolve this comment.

      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
```
[//]: # ({x-version-update-end})

#### Choosing your implementation

If you are using Maven, add this to your pom.xml file (notice that you can replace
`google-auth-library-oauth2-http` with any of `google-auth-library-credentials` and
`google-auth-library-appengine`, depending on your application needs):

```xml
<dependency>
  <groupId>com.google.auth</groupId>
  <!-- Let the BOM manage the transitive dependencies and module version. -->

>   <!-- Replace with the implementation module that suits your needs -->

Reviewer: ```suggestion
  <!-- Replace with the module(s) that are needed -->
```

  <artifactId>google-auth-library-oauth2-http</artifactId>
</dependency>
```

### Using Gradle
If you are using Gradle, add this to your dependencies

[//]: # ({x-version-update-start:google-auth-library-bom:released})
```Groovy
dependencies {
    // The BOM will manage the module versions and transitive dependencies

>     implementation platform('com.google.auth:google-auth-library-bom:1.30.1')

Reviewer: can this version be updated in release-please?

Author: Yes, I recently fixed the readme updates in https://github.com/googleapis/google-auth-library-java/pull/1616 and confirmed it works in https://github.com/googleapis/google-auth-library-java/pull/1618/files

Reviewer: Awesome, thank you!

>     // Replace this with the implementation module that suits your needs

Reviewer: ```suggestion
    // Replace with the module(s) that are needed
```

    implementation 'com.google.auth:google-auth-library-oauth2-http'
}

```
[//]: # ({x-version-update-end})

Unfortunately, SBT [cannot](https://github.com/sbt/sbt/issues/4531) manage dependencies via Maven
Bills of Materials. Therefore, you will have to add the submodule directly. Make sure the module
versions are aligned in case you are using more than one authentication module in order to prevent
transitive dependency conflicts.
If you are using SBT, add this to your dependencies

[//]: # ({x-version-update-start:google-auth-library-oauth2-http:released})
```Scala
// Replace this with the implementation module that suits your needs

> libraryDependencies += "com.google.auth" % "google-auth-library-oauth2-http" % "1.30.1"

Reviewer: Can you double check that this version can also be updated?

Author: Same as in https://github.com/googleapis/google-auth-library-java/pull/1620/files#r1920682167

## Quickstart

> ### Alternative: using `java-libraries-bom`

Reviewer: Perhaps not as `alternative` since it's the first option listed. I think we can probably put something like `Preferred/ Preference`?

  </dependencies>
</dependencyManagement>
```

Otherwise, if you don't plan using libraries-bom, see the next section on 
_Google Auth Library Bill of Materials_.

### Using Maven

#### Google Auth Library Bill of Materials
In order to ensure transitive dependencies and the modules themselves are aligned with each other,
we rely on the Google Auth Library Bill of Materials. Please add this to your dependency management
section as follows:

[//]: # ({x-version-update-start:google-auth-library-bom:released})
```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>com.google.auth</groupId>
      <artifactId>google-auth-library-bom</artifactId>
      <version>1.30.1</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
```
[//]: # ({x-version-update-end})

#### Choosing your implementation

If you are using Maven, add this to your pom.xml file (notice that you can replace
`google-auth-library-oauth2-http` with any of `google-auth-library-credentials` and
`google-auth-library-appengine`, depending on your application needs):

```xml
<dependency>
  <groupId>com.google.auth</groupId>

>   <!-- Let the BOM manage the transitive dependencies and module version. -->

Reviewer: ```suggestion
  <!-- Let the BOM manage the module and dependency versions -->
```
